### DISCUSSION TITLE
InstantSearch React ToggleRefinement off parameter

### DISCUSSION THREAD

**Discussion by RichardJohnn** [algolia/instantsearch#4474] [General] [answered]
Looking at https://github.com/algolia/instantsearch.js/issues/905 I see a solution for setting the `off` value to explicitly be `false` for a ToggleRefinement.  However, for the React component https://www.algolia.com/doc/api-reference/widgets/toggle-refinement/react/#widget-param-value it seems you can only set `value` which is the value used when checked. 

Is there a way to set the unchecked, `off` value with the React instant search component?

Thank you

**Comment by Haroenv** [comment, marked as answer]
Hi @RichardJohnn, this isn't currently an option for React InstantSearch. Since React InstantSearch does not use InstantSearch behind the hood, adding support for that needs to be manually added, by adding an else clause to 

https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-core/src/connectors/connectToggleRefinement.js#L156

and in that clause add a disjunctive refinement for the "off value". You can do this either by forking, using patch-package or copying the implementation into your code base.

Since this has been a feature for a while in InstantSearch, but has never been in React InstantSearch, could you open an issue there so we can gauge interest by the number of upvotes on the ticket?

Thanks!

**Reply by RichardJohnn** [reply]
thank you for the tip!
also opened: https://github.com/algolia/react-instantsearch/issues/2963
